tiledb/sm/query/readers/attribute_order_validator.h

```diff
@@ -0,0 +1,581 @@
+      }
+
+      // If not validated and fragment to check is not yet found, check if
+      // this fragment is over-lapping or directly following the max.
+      if (!finished_upper_search && (in_domain(max, non_empty_domain2) ||
+                                     max + 1 == non_empty_domain2[0])) {
+        frag_to_compare.second = f2;
+        finished_upper_search = true;
+      }
+
+      // If both fragments are either validated or the matching fragment was
+      // found, then we are done searching.
+      if (finished_lower_search && finished_upper_search) {
+        return;
+      }
+    }
+
+    // If the search/validation failed, then there is  a discontinuity in this
+    // array.
+    throw AttributeOrderValidatorStatusException(
+        "Discontiuity found in array domain");
+  }
+
+  /**
+   * Performs validation that can be done without loading tile.
+   *
+   * If a validation check fails, this will throw an error.
+   *
+   * @tparam Index type.
+   * @tparam Attribute type.
+   * @param index_dim The index dimension.
+   * @param f Fragment index.
+   * @param non_empty_domains Vector of pointers to the non-empty domain for
+   *     each fragment.
+   *
+   */
+  template <typename IndexType, typename AttributeType>
+  void validate_without_loading_tiles(
+      const ArraySchema& schema,
+      const Dimension* index_dim,
+      bool increasing_data,
+      uint64_t f,
+      const std::vector<const void*>& non_empty_domains,
+      const std::vector<shared_ptr<FragmentMetadata>> fragment_metadata,
+      const std::vector<uint64_t>& frag_first_array_tile_idx) {
+    // For easy reference.
+    const IndexType* non_empty_domain =
+        static_cast<const IndexType*>(non_empty_domains[f]);
+    const IndexType* dim_dom = index_dim->domain().typed_data<IndexType>();
+    auto tile_extent{index_dim->tile_extent().rvalue_as<IndexType>()};
+
+    if (!min_validated_[f]) {
+      // Get the fragment number to compare against.
+      if (!fragment_to_compare_against_[f].first.has_value()) {
+        throw std::logic_error(
+            "Should know fragment value to compare for non-validated bound.");
+      }
+      auto f2 = fragment_to_compare_against_[f].first.value();
+
+      // Get the min index. See if the min is tile aligned.
+      auto min = non_empty_domain[0];
+      bool min_tile_aligned = min == index_dim->round_to_tile<IndexType>(
+                                         min, dim_dom[0], tile_extent);
+
+      // Get the tile index of the tile in f2 immediately proceeding f (with
+      // respect to tile index in f2). If f is tile aligned we need to subtract
+      // an additional value.
+      uint64_t f2_tile_idx = frag_first_array_tile_idx[f] -
+                             frag_first_array_tile_idx[f2] - min_tile_aligned;
+
+      // If we are tile aligned or the min is right next to the other
+      // fragment's max, we can validate. Otherwise we'll need to load
+      // the tile.
```

Comment:
```suggestion
      // If we are tile aligned or non-overlapping (the min is right next to
      // the other fragment's max), we can validate. Otherwise, we'll need to 
      // load the tile.
```